iam.tf

```diff
@@ -1,7 +1,10 @@
 locals {
   irsa_role_create = var.enabled && var.service_account_create && var.irsa_role_create
+  partition        = data.aws_partition.current.id
 }
 
+data "aws_partition" "current" {}
```

Comment:
Hello @sclem, thanks for your contribution! Adding support for other AWS partitions is a welcomed change. But I think the module should be more explicit about it, meaning what I would like to do instead of automatic partition detection via `data.aws_partition` is to pass it explicitly via variable instead, i.e. 
```
variable "aws_partition" {
  type        = string
  default     = "aws"
  description = "AWS partition in which the resources are located. Available values are `aws`, `aws-cn`, `aws-us-gov`"
}
```
Let me know your thoughts. Thanks.

Reply:
Hello @sclem, are you willing to take this further or whats your view on the matter, please?